File: Task-03/prime.cpp
#include <iostream>

// Function to check whether a number is prime or not
bool isPrime(int num) {
    if (num <= 1) {
        return false;  // Not a prime number
    }

    for (int i = 2; i * i <= num; i++) {
        if (num % i == 0) {
            return false;  // Not a prime number
        }
    }

    return true;  // Prime number
}

int main() {
    int number;

    // Input a number from the user
    std::cout << "Enter a number: ";
    std::cin >> number;

    // Check if the number is prime or not
    if (isPrime(number)) {
        std::cout << number << " is a prime number." << std::endl;
    } else {
        std::cout << number << " is not a prime number." << std::endl;
    }

    return 0;
}

File: Task-03/prime.rs
use std::io;

// Function to check whether a number is prime or not
fn is_prime(num: u32) -> bool {
    if num <= 1 {
        return false;  // Not a prime number
    }

    for i in 2..=(num as f64).sqrt() as u32 {
        if num % i == 0 {
            return false;  // Not a prime number
        }
    }

    true  // Prime number
}

fn main() {
    // Input a number from the user
    println!("Enter a number: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");

    let number: u32 = input.trim().parse().expect("Please enter a valid number");

    // Check if the number is prime or not
    if is_prime(number) {
        println!("{} is a prime number.", number);
    } else {
        println!("{} is not a prime number.", number);
    }
}
